### ISSUE TITLE
BUG: Sidecar scope one domain, but can access all domains in the virtualservice

### ISSUE THREAD

**Issue by liuzheng** [istio/istio#26457]
**Bug description**
Sidecar scope one domain, but can access all domains in the virtualservice.
In the following example, the pod can access `qq.com`, which is not expected to happen.

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
only access the host in Sidecar hosts.

**Steps to reproduce the bug**
```
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: ubuntu
spec:
  selector:
    matchLabels:
      app: ubuntu
  template:
    metadata:
      labels:
        app: ubuntu
    spec:
      restartPolicy: Always
      containers:
      - name: ubuntu
        image: ubuntu
        imagePullPolicy: Always
        command:
          - sleep
          - "3600"
---
apiVersion: networking.istio.io/v1alpha3
kind: Sidecar
metadata:
  name: ubuntu
spec:
  workloadSelector:
    labels:
      app: ubuntu
  egress:
  - captureMode: IPTABLES
    hosts:
    - "*/archive.ubuntu.com"
  outboundTrafficPolicy:
    mode: REGISTRY_ONLY
---
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: ubuntu
spec:
  hosts:
    - "archive.ubuntu.com"
  ports:
  - number: 443
    name: tls
    protocol: TLS
  - number: 80
    name: http
    protocol: HTTP
  location: MESH_EXTERNAL
  resolution: DNS
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: ubuntu
spec:
  hosts:
    - "*.ubuntu.com"
    - "qq.com"
  http:
  - route:
    - destination:
        host: http-proxy.proxy.svc.cluster.local
        port:
          number: 8080
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
$ istioctl version --remote
client version: 1.7.0-beta.1
control plane version: unknown
data plane version: 1.7.0-beta.1 (3 proxies)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-17T11:33:59Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

**How was Istio installed?**
istioctl

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by ramaraochavali** [comment]
https://github.com/istio/istio/pull/20408 - Based on this your virtual service destinations are also added as clusters

**Comment by liuzheng** [comment]
tested, fixed in 1.7.0-rc.1
